app/frontend/src/features/connections/friends/FriendList.tsx

```diff
@@ -46,52 +43,33 @@ const useStyles = makeStyles((theme) => ({
 function FriendList() {
   const baseClasses = useFriendsBaseStyles();
   const classes = useStyles();
-  const isMounted = useIsMounted();
-  const [errorMessage, setErrorMessage] = useSafeState(isMounted, "");
-  const [loading, setLoading] = useSafeState(isMounted, false);
-  const [friendIds, setFriendIds] = useSafeState<number[]>(isMounted, []);
-  const dispatch = useAppDispatch();
-  const users = useTypedSelector(getUsers);
-
-  useEffect(() => {
-    setLoading(true);
-    (async () => {
-      try {
-        const friendIds = await service.api.listFriends();
-        unwrapResult(await dispatch(fetchUsers({ userIds: friendIds })));
-        setFriendIds(friendIds);
-        setLoading(false);
-      } catch (error) {
-        setErrorMessage(error.message);
-      }
-    })();
-  }, [dispatch, setFriendIds, setErrorMessage, setLoading]);
+  const { errors, isLoading, isError, friendQueries } = useFriendList();
 
   return (
     <Card>
       <Box className={classes.container}>
         <Typography className={baseClasses.header} variant="h2">
           Your friends
         </Typography>
-        {loading ? (
+        {isLoading || friendQueries.some((query) => query.isLoading) ? (
           <CircularProgress className={baseClasses.circularProgress} />
-        ) : errorMessage ? (
+        ) : isError ? (
           <Alert className={baseClasses.errorAlert} severity="error">
-            {errorMessage}
+            {errors.join("\n")}
           </Alert>
         ) : (
-          friendIds.map((friendId) => {
-            const friend = users[friendId]?.user;
+          friendQueries.map((payload) => {
+            const user = payload.data as User.AsObject;
```

Comment:
Can we avoid `as`?

Reply:
Sadly not in this case, since `useQueries` is a generic function that accepts any type of queries, the lib doesn't provide a good way to type it and I keep on getting errors of `unknown` is not assignable to `User.AsObject` when I try to type `payload` instead...

Reply:
I see, how about moving the type assertion into the hook instead? Only if it's easy.

In an closed stale issue they mention they might accept a PR they puts generic types in, I might try that. Could be straightforward.

Reply:
Aha it's a bit of a hack but I managed to augment/extend the `react-query` types to allow `useQueries` to be typed with generics, so no more `as` casting/type assertions!